src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -118,19 +117,18 @@ private static void CallPropertyAddRange<TElement>(object target, object source)
 
         private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
         {
-            var propertyHelpers = PropertyHelper.GetProperties(controller);
+            var controllerMetadata = _modelMetadataProvider.GetMetadataForType(containerType);
```

Comment:
Note that you can still do some of this cleanup by doing `GetMetadataForType(...)`